crates/top/rerun_c/src/lib.rs

```diff
@@ -826,6 +910,113 @@ pub unsafe extern "C" fn rr_recording_stream_log_file_from_contents(
     }
 }
 
+#[allow(unsafe_code)]
+#[allow(clippy::result_large_err)]
+fn rr_recording_stream_send_columns_impl(
+    stream: CRecordingStream,
+    entity_path: CStringView,
+    time_columns: &[CTimeColumn],
+    component_batches: &[CPartitionedComponentBatch],
+) -> Result<(), CError> {
+    // Create chunk-id as early as possible. It has a timestamp and is used to estimate e2e latency.
+    let id = ChunkId::new();
+
+    let stream = recording_stream(stream)?;
+    let entity_path = entity_path.as_str("entity_path")?;
+
+    let time_columns: BTreeMap<Timeline, TimeColumn> = time_columns
+        .iter()
+        .map(|time_column| {
+            let timeline: Timeline = time_column.timeline.clone().try_into()?;
+            let datatype = arrow2::datatypes::DataType::Int64;
+            let time_values_untyped = unsafe { arrow_array_from_c_ffi(&time_column.times, datatype) }?;
+            let time_values = time_values_untyped
+                .as_any()
+                .downcast_ref::<arrow2::array::PrimitiveArray<i64>>()
+                .ok_or_else(|| {
+                    CError::new(
+                        CErrorCode::ArrowFfiArrayImportError,
+                        "Arrow C FFI import did not produce a Int64 time array - please file an issue if you see this! This shouldn't be possible since conversion from C was successful with this datatype.",
```

Comment:
```suggestion
                        "Arrow C FFI import did not produce a Int64 time array - please file an issue at https://github.com/rerun-io/rerun/issues if you see this! This shouldn't be possible since conversion from C was successful with this datatype.",
```